diff --git a/src/Entity/Pet.php b/src/Entity/Pet.php
index 942c7ff..9cb7608 100644
--- a/src/Entity/Pet.php
+++ b/src/Entity/Pet.php
@@ -132,7 +132,7 @@ class Pet extends ContentEntityBase implements PetInterface {
       ->setDisplayConfigurable('form', TRUE)
       ->setDisplayConfigurable('view', TRUE);
 
-    $fields['mail_body'] = BaseFieldDefinition::create('string_long')
+    $fields['mail_body'] = BaseFieldDefinition::create('text_long')
       ->setLabel(t('Mail Body'))
       ->setDescription(t('The body of the email template. May include tokens of any token type specified below.'))
       ->setDefaultValue(NULL)
@@ -146,7 +146,7 @@ class Pet extends ContentEntityBase implements PetInterface {
       ->setDisplayConfigurable('form', TRUE)
       ->setDisplayConfigurable('view', TRUE);
 
-    $fields['mail_body_plain'] = BaseFieldDefinition::create('string_long')
+    $fields['mail_body_plain'] = BaseFieldDefinition::create('text_long')
       ->setLabel(t('Mail Body Plain'))
       ->setDescription(t('The plain text body of the email template. May include tokens of any token type specified below. If left empty Mime Mail will use drupal_html_to_text() to create a plain text version of the email.'))
       ->setDisplayOptions('form', array(
diff --git a/src/Form/PetPreviewForm.php b/src/Form/PetPreviewForm.php
index 4dc5375..55abb98 100644
--- a/src/Form/PetPreviewForm.php
+++ b/src/Form/PetPreviewForm.php
@@ -272,8 +272,8 @@ class PetPreviewForm extends FormBase {
 
     $token = \Drupal::token();
     $storage['subject_preview'] = $token->replace($values['subject'], $subs);
-    $storage['body_preview'] = $token->replace(pet_isset_or($values['mail_body']), $subs);
-    $storage['body_preview_plain'] = $token->replace(pet_isset_or($values['mail_body_plain']), $subs);
+    $storage['body_preview'] = $token->replace(pet_isset_or($values['mail_body']['value']), $subs);
+    $storage['body_preview_plain'] = $token->replace(pet_isset_or($values['mail_body_plain']['value']), $subs);
     $form_state->setStorage($storage);
   }
 }
