diff --git a/src/Element/Captcha.php b/src/Element/Captcha.php
index 6f441a77c..6c21c08a9 100644
--- a/src/Element/Captcha.php
+++ b/src/Element/Captcha.php
@@ -108,7 +108,7 @@ public static function processCaptchaElement(&$element, FormStateInterface $form
     // If there is a submitted form: try to retrieve and reuse the
     // CAPTCHA session ID from the posted data.
     list($posted_form_id, $posted_captcha_sid) = _captcha_get_posted_captcha_info($element, $form_state, $this_form_id);
-    if ($this_form_id == $posted_form_id && isset($posted_captcha_sid)) {
+    if ($this_form_id == $posted_form_id && isset($posted_captcha_sid) && static::captchaSessionExists($posted_captcha_sid)) {
       $captcha_sid = $posted_captcha_sid;
     }
     else {
@@ -248,4 +248,19 @@ public static function processCaptchaElement(&$element, FormStateInterface $form
     return $element;
   }
 
+  /**
+   * @param $csid
+   *
+   * @return bool
+   */
+  protected static function captchaSessionExists($csid) {
+    $session = \Drupal::database()->select('captcha_sessions', 'cs')
+      ->fields('cs', ['csid'])
+      ->condition('csid', $csid)
+      ->execute()
+      ->fetchAll();
+
+    return !empty($session);
+  }
+
 }
