diff --git a/core/misc/states.es6.js b/core/misc/states.es6.js
index dd2c315..f5069c9 100644
--- a/core/misc/states.es6.js
+++ b/core/misc/states.es6.js
@@ -202,6 +202,10 @@
      *   true or false.
      */
     compare(reference, selector, state) {
+      if (typeof this.values[selector] === 'undefined' || typeof this.values[selector][state.name] === 'undefined') {
+        return false;
+      }
+
       const value = this.values[selector][state.name];
       if (reference.constructor.name in states.Dependent.comparisons) {
         // Use a custom compare function for certain reference value types.
diff --git a/core/misc/states.js b/core/misc/states.js
index fcdc37a..5b0eed4 100644
--- a/core/misc/states.js
+++ b/core/misc/states.js
@@ -98,6 +98,10 @@
       });
     },
     compare: function compare(reference, selector, state) {
+      if (typeof this.values[selector] === 'undefined' || typeof this.values[selector][state.name] === 'undefined') {
+        return false;
+      }
+
       var value = this.values[selector][state.name];
       if (reference.constructor.name in states.Dependent.comparisons) {
         return states.Dependent.comparisons[reference.constructor.name](reference, value);
