diff --git a/linkit/src/Plugin/Filter/LinkitFilter.php b/linkit/src/Plugin/Filter/LinkitFilter.php
index 11f3e0512..6a9ca6e60 100644
--- a/linkit/src/Plugin/Filter/LinkitFilter.php
+++ b/linkit/src/Plugin/Filter/LinkitFilter.php
@@ -98,6 +98,7 @@ public function process($text, $langcode) {
             $substitution_type = $entity_type === 'file' ? 'file' : SubstitutionManagerInterface::DEFAULT_SUBSTITUTION;
           }
 
+          if ($entity_type) {
           $entity = $this->entityRepository->loadEntityByUuid($entity_type, $uuid);
           if ($entity) {
 
@@ -127,6 +128,7 @@ public function process($text, $langcode) {
               ->addCacheableDependency($entity);
           }
         }
+        }
         catch (\Exception $e) {
           watchdog_exception('linkit_filter', $e);
         }
