diff --git a/google_maps_api.module b/google_maps_api.module
index c97dd4c..4efc53e 100644
--- a/google_maps_api.module
+++ b/google_maps_api.module
@@ -14,7 +14,7 @@ function google_maps_api_library_info_build() {
     'version' => '1.x',
     'js' => [
       $url => [
-        'type' => 'external', 'attributes' => [ 'defer' => 'true', 'async' => 'true' ]
+        'type' => 'external', 'attributes' => [ 'defer' => TRUE, 'async' => TRUE ]
       ]
     ]
   ];
