diff --git a/src/RoleHierarchyHelper.php b/src/RoleHierarchyHelper.php
index ca99123..f87184d 100644
--- a/src/RoleHierarchyHelper.php
+++ b/src/RoleHierarchyHelper.php
@@ -45,6 +45,9 @@ public static function getAccountRoleWeight(AccountInterface $account) {
    *   The weight of the role.
    */
   public static function getRoleWeight($role) {
+    if ($role == 'workflow_author') {
+      return 9999;
+    }
     return Role::load($role)->getWeight();
   }
 
